streams/src/main/java/org/apache/kafka/streams/state/internals/MemoryLRUCacheBytes.java

```diff
@@ -0,0 +1,437 @@
+            final Bytes key = tail.key;
+            final String namespace = tail.namespace;
+            final TreeMap<Bytes, LRUNode> treeMap = this.map.get(namespace);
+            final LRUNode toRemove = treeMap.get(key);
+            currentSizeBytes -= toRemove.size();
+            if (toRemove.entry.isDirty()) {
+                callListener(toRemove.key, toRemove);
+            }
+            remove(toRemove);
+            treeMap.remove(key);
+
+            if (toRemove.entry.isDirty()) {
+                removeDirtyKey(namespace, key);
+            }
+        }
+    }
+
+
+    public synchronized void put(final String namespace, byte[] key, MemoryLRUCacheBytesEntry value) {
+        final Bytes cacheKey = cacheKey(key);
+        LRUNode node;
+        // get map
+        if (!this.map.containsKey(namespace)) {
+            this.map.put(namespace, new TreeMap<Bytes, LRUNode>());
+        }
+
+        final TreeMap<Bytes, LRUNode> treeMap = this.map.get(namespace);
+
+        if (treeMap.containsKey(cacheKey)) {
+            node = treeMap.get(cacheKey);
+            currentSizeBytes -= node.size();
+            node.update(value);
+            updateLRU(node);
+        } else {
+            node = new LRUNode(cacheKey, namespace, value);
+            // put element
+            putHead(node);
+            treeMap.put(cacheKey, node);
+        }
+        if (value.isDirty()) {
+            addDirtyKey(namespace, cacheKey);
+        }
+
+        currentSizeBytes += node.size();
+        maybeEvict();
+    }
+
+    public synchronized MemoryLRUCacheBytesEntry putIfAbsent(final String namespace, byte[] key, MemoryLRUCacheBytesEntry value) {
+        final MemoryLRUCacheBytesEntry originalValue = get(namespace, key);
+        if (originalValue == null) {
+            put(namespace, key, value);
+        }
+        return originalValue;
+    }
+
+    public synchronized void putAll(final String namespace, final List<KeyValue<byte[], MemoryLRUCacheBytesEntry>> entries) {
+        for (KeyValue<byte[], MemoryLRUCacheBytesEntry> entry : entries) {
+            put(namespace, entry.key, entry.value);
+        }
+    }
+
+    public synchronized MemoryLRUCacheBytesEntry delete(final String namespace, final byte[] key) {
+        final Bytes cacheKey = cacheKey(key);
+        final TreeMap<Bytes, LRUNode> treeMap = this.map.get(namespace);
+        if (treeMap == null) {
+            return null;
+        }
+
+        final LRUNode node = treeMap.get(cacheKey);
+
+        if (node == null) {
+            return null;
+        }
+
+        // remove from LRU list
+        remove(node);
+
+        // remove from map
+        final LRUNode value = treeMap.remove(cacheKey);
+        if (treeMap.size() == 0) {
+            this.map.remove(namespace);
+        }
+        removeDirtyKey(namespace, cacheKey);
+        return value.entry();
+    }
+
+
+    public synchronized int dirtySize(final String namespace) {
```

Comment:
I think the question is:  Shouldn't we rather use `long` rather than `int` throughout?